src/simulators/tensor_network/matrix_product_state.hpp

```diff
@@ -191,8 +193,23 @@ class MPS{
     return 0;
   }
   
+  //----------------------------------------------------------------
+  // function name: update_qubit_and_neighbor
+  // Description: This function is used by apply_measure to propagate
+  // the effects of measurement from the measured qubit to all other
+  // qubits
+  //----------------------------------------------------------------
+  void update_qubit_and_neighbor(uint_t index, 
+				 Direction direction,
+				 const cmatrix_t &mul_matrix);
   reg_t sample_measure(std::vector<double> &rnds);
 
+  reg_t apply_measure(const reg_t &qubits, 
+		      RngEngine &rng);
```

Comment:
```suggestion
	               RngEngine &rng) const;
```

Reply:
This function can be `const` as seems it's not changing the state of the object

Reply:
apply_measure does change the state, so I can not turn it into 'const'.
sample_measure could in fact be const, but I removed it completely because it is implemented by saving the internal state and then using apply_measure on it.